### PR TITLE
Add scalafix.

### PR DESCRIPTION
**Pull request by mwz**
I added scalafix in an effort to try to improve the overall quality of the codebase. I didn't address any of the issues yet, but adding scalafix is a good start.

There are a lot of instances of using vars, throws, nulls and universal equality, so I disabled those rules in the config as they would make scalafix fail (you can turn those on and see the results by running `fix` in sbt.) I didn't find a way to report those errors as warnings, so I'll probably raise an issue in scalafix so this can be implemented in the future. The goal is to turn on those disabled rules once all the issues have been resolved.

Another benefit of scalafix is automatic sorting of imports for users of metals (via the [NeQuissimus/sort-imports](https://github.com/NeQuissimus/sort-imports) external rule.), which I also added to the config. I created some sbt aliases too:
- `fix` runs scalafix and sorts imports
- `fixCheck` will be useful for running in PR checks to ensure there are no new issues and the imports are already sorted. I didn't integrate it with travis, because I'm hoping to add this straight to github actions instead.

You can also run the sort imports commands separately (`fixImports` and `fixCheckImports`), but I usually just run `fix`. I added them mostly so that I can reference them in the `fix` and `fixCheck` commands.

### CODE REVIEW

File: src/test/scala/com/sksamuel/scapegoat/inspections/equality/ComparingUnrelatedTypesTest.scala [hunk 1]
```diff
@@ -33,73 +32,6 @@ class ComparingUnrelatedTypesTest
         compileCodeSnippet(code)
         compiler.scapegoat.feedback.warnings.size shouldBe 1
       }
-      "for multiple comparisons" in {
```
Comment (mccartney) [inline src/test/scala/com/sksamuel/scapegoat/inspections/equality/ComparingUnrelatedTypesTest.scala]: Why is that being removed?

Comment (mwz) [reply src/test/scala/com/sksamuel/scapegoat/inspections/equality/ComparingUnrelatedTypesTest.scala]: So this is dead code if you scroll further down.

File: src/test/scala/com/sksamuel/scapegoat/inspections/VarCouldBeValTest.scala [hunk 2]
```diff
@@ -59,13 +59,16 @@ class VarCouldBeValTest extends AnyFreeSpec with Matchers with PluginRunner with
...
-        bazWarning.snippet.exists(_.contains("var baz: Int = 3")) shouldBe true
+
+        warningsInOrder.size shouldBe 2
+        warningsInOrder match {
+          case Seq(barWarning, bazWarning) =>
+            barWarning.line shouldBe 3
+            barWarning.snippet.exists(_.contains("var bar: Int = 1")) shouldBe true
+            bazWarning.line shouldBe 5
+            bazWarning.snippet.exists(_.contains("var baz: Int = 3")) shouldBe true
+        }
```
Comment (mwz) [inline src/test/scala/com/sksamuel/scapegoat/inspections/VarCouldBeValTest.scala]: Minor refactoring to make scalafix happy:
>  Pattern matching in val assignment can result in match error, use "_ match { ... }" w
ith a fallback case instead.

File: src/test/scala/com/sksamuel/scapegoat/inspections/equality/ComparingUnrelatedTypesTest.scala (hunk 1 above)
Comment (mccartney) [reply src/test/scala/com/sksamuel/scapegoat/inspections/equality/ComparingUnrelatedTypesTest.scala]: Right!

**Pull request merged**